diesel_cli/src/database.rs

```diff
@@ -134,38 +136,63 @@ impl InferConnection {
             Backend::Sqlite => SqliteConnection::establish(&database_url).map(Self::Sqlite),
         };
 
-        result.unwrap_or_else(|err| handle_error_with_database_url(&database_url, err))
+        let mut conn = result.map_err(|err| crate::errors::Error::ConnectionError {
+            error: err,
+            url: database_url,
+        })?;
+
+        conn.set_instrumentation(|event: InstrumentationEvent<'_>| {
+            if let InstrumentationEvent::FinishQuery { query, error, .. } = event {
+                if let Some(err) = error {
+                    tracing::error!(?query, ?err, "Failed to execute query");
+                } else {
+                    tracing::debug!(?query);
+                }
```

Comment:
Should this be opt in? Sometimes queries can have sensitive data, so I wonder if clients using `diesel` should be able to configure the raw SQL being logged or not.

Reply:
This change only affects diesel-cli. It contains quite a few hardcoded queries for loading schema information. The only user provided SQL that's executed there is the SQL in your migrations. Given that you run the client tool mostly manually I wouldn't expect that to leak any important information.